some docs: osx vmImage, sdks, and metal (#1638)
conda-forge/conda-forge.github.io: src/maintainer/knowledge_base.rst

Requiring newer macOS SDKs
==========================

> At conda-forge,we use macOS SDK 10.9 to build software so that they can be deployed to

Reviewer: ```suggestion
At conda-forge, we use macOS SDK 10.9 to build software so that they can be deployed to
```

the requirement should be changed from ``run`` to ``run_constrained``. Note that
``conda<4.8`` will ignore the condition if it's a ``run_constrained`` on ``__osx``.

Compile for Apple's Metal framework
-----------------------------------

It is possible to compile against Apple's Metal framework. 
However, you will likely need to use newer SDKs (11 and newer) for functionality.
If in doubt, you can search the `phracker/MacOSX-SDKs <https://github.com/phracker/MacOSX-SDKs.git>`__ 
repo for the exact symbols and their availability.

Changing the Azure vmImage
--------------------------

Currently, the conda-forge's default vmImage for macOS is 10.15.
If you increase the ``MACOSX_DEPLOYMENT_TARGET`` beyond this, the 
``__osx`` run condition will prevent you from running tests.

> To use a newer vmImage, you can do it by editing ``conda-forge.yml`` 

Reviewer: ```suggestion
To use a newer vmImage, edit ``conda-forge.yml`` 
```

> in the root of the repo like below. Note that the release after 10.15 is 11, 

Reviewer: ```suggestion
in the root of the repo like below. 
Note that the release after 10.15 is 11, 
```
I think a note should be slightly more prominent (Haven't checked though if this actually renders a linebreak).

Author: I will try to rerender the whole thing to check, thanks for the valuable comments!

but Azure pipelines only offer the latest of 11, ``macOS-11``, which 
at the time of writing (March 2022) is 11.6. (Currently, the ``macOS-12`` vmImage is in 

> private preview.)

Reviewer: Maybe add a link here to https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops&tabs=yaml#software

Author: Also, as far as I could tell, specifying the `vmImage` has no real benefits other than this narrow case above.

.. code-block:: yaml

    azure:
      settings_osx:
        pool:
          vmImage: macOS-11

> As always, remember to rerender for changes to take effect.

Reviewer: I would add a link to the rerendering section of the docs. Should be under infrastructure.

the requirement should be changed from ``run`` to ``run_constrained``. Note that
``conda<4.8`` will ignore the condition if it's a ``run_constrained`` on ``__osx``.

Compile for Apple's Metal framework
-----------------------------------

It is possible to compile against Apple's Metal framework. 
However, you will likely need to use newer SDKs (11 and newer) for functionality.
If in doubt, you can search the `phracker/MacOSX-SDKs <https://github.com/phracker/MacOSX-SDKs.git>`__ 
repo for the exact symbols and their availability.

> 

Reviewer 2: Maybe pyobjc-framework-metal can be used as an example?
https://github.com/conda-forge/pyobjc-framework-metal-feedstock/blob/main/recipe/conda_build_config.yaml
